src/main/java/com/datatheorem/mobileappsecurity/jenkins/plugin/sendbuild/SendBuildAction.java

```diff
@@ -398,6 +400,21 @@ HttpResponse uploadBuildRequest()
         }
         listener.getLogger().println("Build file path is: " + buildPath);
 
+        // Add credential username
+        if (getApplicationCredentialUsername() != null && !getApplicationCredentialPassword().isEmpty()){
```

Comment:
At this point, it has not been verified that `getApplicationCredentialPassword()` is not `null`; I think the expected thing would be to add an username and a password only if both are provided.